components/public-api-server/middleware/logging.go

```diff
@@ -0,0 +1,30 @@
+package middleware
+
+import (
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"time"
+)
+
+type Middleware func(handler http.Handler) http.Handler
+
+func NewLoggingMiddleware(l *logrus.Entry) Middleware {
+
+	return func(next http.Handler) http.Handler {
+		logging := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			start := time.Now()
+			uri := r.RequestURI
+			method := r.Method
+			duration := time.Since(start)
+			next.ServeHTTP(w, r)
+
+			l.WithFields(logrus.Fields{
+				"uri":      uri,
+				"method":   method,
+				"duration": duration,
+			}).Info("Method call successful")
```

Comment:
```suggestion
			}).Infof("Handled HTTP request %s %s", method, uri)
```

By also adding the method and URI into the log, we can more easily visually grep and we still have it in the structured fields for deeper analysis. 
It also changes from `succesful` because that's somewhat ambigous. The fact we handled it doesn't mean we returned a 200 (or something that can be consider a success) so that's the reasoning for the text change.

Reply:
This makes sense, thanks!